packages/flutter/lib/src/material/button_style_button.dart

```diff
@@ -389,6 +389,16 @@ class _ButtonStyleState extends State<ButtonStyleButton> with TickerProviderStat
       });
     }
 
+    Color? effectiveIconColor() {
+      return widgetStyle?.iconColor?.resolve(statesController.value) ??
+          themeStyle?.iconColor?.resolve(statesController.value) ??
+          widgetStyle?.foregroundColor?.resolve(statesController.value) ??
+          themeStyle?.foregroundColor?.resolve(statesController.value) ??
+          defaultStyle.iconColor?.resolve(statesController.value) ??
+          // Fallback to foregroundColor if iconColor is null.
+          defaultStyle.foregroundColor?.resolve(statesController.value);
+    }
+
```

Comment:
There are multiple issues closed due to the regression this PR fixes. @rydmike suggested we should CP.  This has low risk changes, mainly the these 6 lines updating the how the icon color is resolved.